From 207a62bff3cce385cb114a79283efcdafa5af438 Mon Sep 17 00:00:00 2001
From: Nithya <nithya.mani@caviumnetworks.com>
Date: Fri, 22 Jan 2016 12:05:45 -0800
Subject: [PATCH 0380/1239] dts : integrate commit
 4ebc7811ca0713cc9478d82f83269f4e19fc3563 from UEFI

Integrate the commit thunderx,dts: Update PEM nodes so we can do PME/AER
to support PCIe/AER

thunderx,dts: Update PEM nodes so we can do PME/AER
Signed-off-by: Nithya <nithya.mani@caviumnetworks.com>
---
 arch/arm/dts/thunder-88xx.dtsi | 39 ++++++++++++++++++++++------------
 1 file changed, 26 insertions(+), 13 deletions(-)

diff --git a/arch/arm/dts/thunder-88xx.dtsi b/arch/arm/dts/thunder-88xx.dtsi
index 0fc2e64374..482037e28d 100644
--- a/arch/arm/dts/thunder-88xx.dtsi
+++ b/arch/arm/dts/thunder-88xx.dtsi
@@ -750,7 +750,8 @@
 			reg = <0x8480 0x00000000 0 0x10000000>;	 /* Configuration space */
 			ranges = <0x03000000 0x8010 0x00000000 0x8010 0x00000000 0x070 0x00000000>, /* mem ranges */
 				 <0x03000000 0x8680 0x00000000 0x8680 0x00000000 0x160 0x24000000>, /* hole for UARTs */
-				 <0x03000000 0x87e0 0x26000000 0x87e0 0x26000000 0x01f 0xda000000>;
+				 <0x03000000 0x87e0 0x26000000 0x87e0 0x26000000 0x000 0x9a000000>, /* hole for PEMs */
+				 <0x03000000 0x87e0 0xc6000000 0x87e0 0xc6000000 0x01f 0x3a000000>;
 
 			mrml-bridge0@1,0 {
 				compatible = "cavium,thunder-8890-mrml-bridge";
@@ -1179,10 +1180,12 @@
 			#size-cells = <2>;
 			#address-cells = <3>;
 			#stream-id-cells = <1>;
-			reg = <0x8800 0x20000000 0x0 0x38000000>;  /* Configuration space */
+			reg = <0x8800 0x20000000 0x0 0x38000000>,  /* Configuration space */
+				<0x87e0 0xc0000000 0x0 0x01000000>; /* PEM space */
 			ranges = <0x01000000 0x00 0x00000000  0x8830 0x00000000  0x00 0x00010000>, /* I/O */
 				 <0x03000000 0x00 0x10000000  0x8810 0x10000000  0x0f 0xf0000000>, /* mem64 */
-				 <0x43000000 0x10 0x00000000  0x8820 0x00000000  0x10 0x00000000>; /* mem64-pref */
+				 <0x43000000 0x10 0x00000000 0x8820 0x00000000 0x10 0x00000000>, /* mem64-pref */
+				 <0x03000000 0x87e0 0xc0000000 0x87e0 0xc0000000 0x00 0x01000000>; /* mem64 PEM */
 
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 7>;
@@ -1201,10 +1204,12 @@
 			#size-cells = <2>;
 			#address-cells = <3>;
 			#stream-id-cells = <1>;
-			reg = <0x8840 0x58000000 0x0 0x38000000>;  /* Configuration space */
+			reg = <0x8840 0x58000000 0x0 0x38000000>,  /* Configuration space */
+				<0x87e0 0xc1000000 0x0 0x01000000>; /* PEM space */
 			ranges = <0x01000000 0x00 0x00010000  0x8870 0x00010000  0x00 0x00010000>, /* I/O */
 				 <0x03000000 0x00 0x10000000  0x8850 0x10000000  0x0f 0xf0000000>, /* mem64 */
-				 <0x43000000 0x10 0x00000000  0x8860 0x00000000  0x10 0x00000000>; /* mem64-pref */
+				 <0x43000000 0x10 0x00000000 0x8860 0x00000000 0x10 0x00000000>, /* mem64-pref */
+				 <0x03000000 0x87e0 0xc1000000 0x87e0 0xc1000000 0x00 0x01000000>; /* mem64 PEM */
 
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 7>;
@@ -1223,10 +1228,12 @@
 			#size-cells = <2>;
 			#address-cells = <3>;
 			#stream-id-cells = <1>;
-			reg = <0x8880 0x90000000 0x0 0x58000000>;  /* Configuration space */
+			reg = <0x8880 0x90000000 0x0 0x58000000>,  /* Configuration space */
+				<0x87e0 0xc2000000 0x0 0x01000000>; /* PEM space */
 			ranges = <0x01000000 0x00 0x00020000  0x88b0 0x00020000  0x00 0x00010000>, /* I/O */
 				 <0x03000000 0x00 0x10000000  0x8890 0x10000000  0x0f 0xf0000000>, /* mem64 */
-				 <0x43000000 0x10 0x00000000  0x88a0 0x00000000  0x10 0x00000000>; /* mem64-pref */
+				 <0x43000000 0x10 0x00000000 0x88a0 0x00000000 0x10 0x00000000>, /* mem64-pref */
+				 <0x03000000 0x87e0 0xc2000000 0x87e0 0xc2000000 0x00 0x01000000>; /* mem64 PEM */
 
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 7>;
@@ -1245,10 +1252,12 @@
 			#size-cells = <2>;
 			#address-cells = <3>;
 			#stream-id-cells = <1>;
-			reg = <0x8900 0x20000000 0x0 0x38000000>;  /* Configuration space */
+			reg = <0x8900 0x20000000 0x0 0x38000000>,  /* Configuration space */
+				<0x87e0 0xc3000000 0x0 0x01000000>; /* PEM space */
 			ranges = <0x01000000 0x00 0x00030000  0x8930 0x00030000  0x00 0x00010000>, /* I/O */
 				 <0x03000000 0x00 0x10000000  0x8910 0x10000000  0x0f 0xf0000000>, /* mem64 */
-				 <0x43000000 0x10 0x00000000  0x8920 0x00000000  0x10 0x00000000>; /* mem64-pref */
+				 <0x43000000 0x10 0x00000000 0x8920 0x00000000 0x10 0x00000000>, /* mem64-pref */
+				 <0x03000000 0x87e0 0xc3000000 0x87e0 0xc3000000 0x00 0x01000000>; /* mem64 PEM */
 
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 7>;
@@ -1267,10 +1276,12 @@
 			#size-cells = <2>;
 			#address-cells = <3>;
 			#stream-id-cells = <1>;
-			reg = <0x8940 0x58000000 0x0 0x38000000>;  /* Configuration space */
+			reg = <0x8940 0x58000000 0x0 0x38000000>,  /* Configuration space */
+				<0x87e0 0xc4000000 0x0 0x01000000>; /* PEM space */
 			ranges = <0x01000000 0x00 0x00040000  0x8970 0x00040000  0x00 0x00010000>, /* I/O */
 				 <0x03000000 0x00 0x10000000  0x8950 0x10000000  0x0f 0xf0000000>, /* mem64 */
-				 <0x43000000 0x10 0x00000000  0x8960 0x00000000  0x10 0x00000000>; /* mem64-pref */
+				 <0x43000000 0x10 0x00000000 0x8960 0x00000000 0x10 0x00000000>, /* mem64-pref */
+				 <0x03000000 0x87e0 0xc4000000 0x87e0 0xc4000000 0x00 0x01000000>; /* mem64 PEM */
 
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 7>;
@@ -1289,10 +1300,12 @@
 			#size-cells = <2>;
 			#address-cells = <3>;
 			#stream-id-cells = <1>;
-			reg = <0x8980 0x90000000 0x0 0x38000000>;  /* Configuration space */
+			reg = <0x8980 0x90000000 0x0 0x38000000>,  /* Configuration space */
+				<0x87e0 0xc5000000 0x0 0x01000000>; /* PEM space */
 			ranges = <0x01000000 0x00 0x00050000  0x89b0 0x00050000  0x00 0x00010000>, /* I/O */
 				 <0x03000000 0x00 0x10000000  0x8990 0x10000000  0x0f 0xf0000000>, /* mem64 */
-				 <0x43000000 0x10 0x00000000  0x89a0 0x00000000  0x10 0x00000000>; /* mem64-pref */
+				 <0x43000000 0x10 0x00000000 0x89a0 0x00000000 0x10 0x00000000>, /* mem64-pref */
+				 <0x03000000 0x87e0 0xc5000000 0x87e0 0xc5000000 0x00 0x01000000>; /* mem64 PEM */
 
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 7>;
-- 
2.29.0

